spec/validators/tags_validator_spec.rb

```diff
@@ -1,46 +1,38 @@
 RSpec.describe TagsValidator do
-  class TagsValidatable
-    include ActiveModel::Validations
-    include ActiveModel::Model
+  let(:record_class) do
+    Class.new do
+      include ActiveModel::Validations
+      include ActiveModel::Model
 
-    attr_accessor :tags
+      attr_accessor :tags
 
-    validates :tags, tags: true
+      validates :tags, tags: true
+    end
   end
 
-  subject(:model) { TagsValidatable.new }
+  before { stub_const("ExampleModel", record_class) }
 
-  context "when valid tags are provided" do
-    before do
-      model.tags = {
-        topics: { any: %w[dogs cats], all: %w[horses] },
-        policies: { any: %w[wWelcome1098-_/], all: %w[news_story] },
-        commodity_type: { any: %w[f3bbdec2-0e62-4520-a7fd-6ffd5d36e03a], all: %w[123-Abc] },
```

Comment:
I found myself rather confused by the granularity of those so I thought it was best to simplify them to a lone test that fails with characters outside the regex. It could have a test that exhausts the regex more fully, but it seemed a simple enough regex that it wasn't particularly beneficial.

Reply:
I almost agree, and the new tests are much clearer. The input that I think would be worth keeping is the UUID, because:

- The `-` (and numbers) isn't covered by the new valid/invalid tests.
- The `-` is quite separate from the alpha regex covered by `dog`.
- UUIDs do turn up as tag values in practice.

I do agree the others don't seem worth checking, as they look like instances where we should just be normalising the data beforehand.

Reply:
> Thanks @benthorner, I've updated the tests for the tags validator to show the items accepted.

Works for me, thanks 👍.